src/main/java/io/fabric8/maven/docker/wait/ExitCodeChecker.java

```diff
@@ -0,0 +1,35 @@
+package io.fabric8.maven.docker.wait;
+
+import io.fabric8.maven.docker.access.DockerAccessException;
+import io.fabric8.maven.docker.service.ServiceHub;
+
+public class ExitCodeChecker implements WaitChecker {
+
+    private final int exitCode;
+    private final ServiceHub hub;
+    private final String containerId;
+
+    public ExitCodeChecker(int exitCode, ServiceHub hub, String containerId) {
+        this.exitCode = exitCode;
+        this.hub = hub;
+        this.containerId = containerId;
+    }
+
+    @Override
+    public boolean check() {
+        try {
+            Integer exitCode = hub.getQueryService().getMandatoryContainer(containerId).getExitCode();
+            if (exitCode == null) {
+                return false;
+            }
+            return exitCode == this.exitCode;
```

Comment:
When a container exited and the exit code does not match the expected exit code, this will never change. So I wonder whether it would not be better to throw an exception in this case? So that the run aborts immediately as it never can meet the wait condition.